tutorials/grails3-send-push-notification-using-firebase-cloud-messaging

```diff
@@ -0,0 +1,196 @@
+
+## Objectives
+
+- Configure Firebase for ability to send push notification via Web Application
+- Create New Grails 3.3.0 Web Application
+- Create function for sending push notification on Grails 3.3.0
+
+## Before You Begin
+
+The following is required for this tutorial:
+
+- Terminal or Command Line
+- Text Editor or IDE ([Atom](https://atom.io/), [Netbeans](https://netbeans.org/))
+- [Java Development Kit 8](http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html)
+- Grails 3.3.0 SDK (https://grails.org/download.html)
+- Google or Gmail account for accessing Firebase dashboard
+
+## Costs
+
+This tutorial uses Firebase Cloud Messaging service which is a free for limited features and capacities.
+
+## Configure Firebase Cloud Messaging
+
+1. Open your browser then go to [Firebase](https://firebase.google.com/).
+2. Login using your Gmail account then click Go to Console button.
+3. Click add project button then give it a name. For this tutorial we give it name "GrailsFCM" then click Create Project button.
+4. It will redirected to Project dashboard. Click gear icon on the left menu then click Project Settings.
+5. Click Cloud Messaging tab then write down Legacy Server Key to your notepad. We will use it later in the Grails web application.
+6. Next configuration is for Android and iOS app. You can find Android configuration [here](https://firebase.google.com/docs/cloud-messaging/android/client) and iOS [here](https://firebase.google.com/docs/cloud-messaging/ios/client).
+
+## Create New Grails 3.3.0 Web Application
+
+We assume that you already installed JDK 8 and Grails 3.3.0 on your computer and added to your environment path. Now, we ready to create the new Grails 3.3.0 application.
+
+1. Open terminal or command line then go to your projects folder.
+2. Type this command.
+   ```
+   grails create-app grails-fcm
+   ```
+3. Go to the newly created Grails 3 application folder.
+   ```
+   cd grails-fcm
+   ```
+4. Enter Grails 3 interactive console by type this command.
+   ```
+   grails
+   ```
+5. Now the you are in Grails interactive console.
+   ```
+   | Resolving Dependencies. Please wait...
+   
+   CONFIGURE SUCCESSFUL
+
+   | Enter a command name to run. Use TAB for completion:
+   grails>
+   ```
+6. Run the Grails web application to make sure everything working properly by type this command in the Grails interactive console.
+   ```
+   run-app
+   ```
+## Add Grails Datastore Rest Client Dependency
+
+We will accessing Firebase using REST API, for that we need to add Grails Datastore Client Rest Dependency in our Grails 3 application. To do that, open and edit build.gradle then add this dependency inside dependencies section. There are two dependencies, one inside buildscript and the other out side buildscripts.
```

Comment:
To access Firebase using REST API, add the Grails Datastore Client Rest dependency to the Grails 3 application. 

1. Open and edit build.gradle.

2. Add the Grails Datastore Client Rest dependency inside the dependencies section. There are two dependencies, one inside buildscript and the other outside buildscripts.